app/services/sessions/retrieve.rb

```diff
@@ -0,0 +1,45 @@
+module Sessions
+  class Retrieve
+    attr_reader :token, :email, :password
+
+    def initialize(token: nil, email: nil, password: nil)
+      @token, @email, @password = token, email, password
+    end
+
+    def process
+      if session.present?
+        get_session
+      else
+        create_session
+      end
```

Comment:
Sure, definitely you don't need to have a service for each controller, actually you can have a service for each controller action if you want, as long as it does the job, and it's SOLID